plugins/chat/app/models/chat_channel.rb

```diff
@@ -107,6 +112,10 @@ def log_channel_status_change(acting_user:)
 
     ChatPublisher.publish_channel_status(self)
   end
+
+  def duplicate_slug?
+    ChatChannel.where(slug: self.slug).where.not(id: self.id).any?
```

Comment:
Do you think we can add a unique index on combination of slug and  chatable_type?